python/graphstorm/gconstruct/construct_graph.py

```diff
@@ -216,25 +217,30 @@ def _process_data(user_pre_parser, user_parser,
         Number of processes to do processing.
     task_info: str
         Task meta info for debugging.
+    ext_mem_workspace : str
+        The path of the external-memory work space.
     """
     if len(two_phase_feat_ops) > 0:
         pre_parse_start = time.time()
-        phase_one_ret = multiprocessing_data_read(in_files, num_proc, user_pre_parser)
+        phase_one_ret = multiprocessing_data_read(in_files, num_proc, user_pre_parser,
+                                                  ext_mem_workspace)
         update_two_phase_feat_ops(phase_one_ret, two_phase_feat_ops)
 
         dur = time.time() - pre_parse_start
         logging.debug("Preprocessing data files for %s takes %.3f seconds.",
                       task_info, dur)
 
     start = time.time()
-    return_dict = multiprocessing_data_read(in_files, num_proc, user_parser)
+    return_dict = multiprocessing_data_read(in_files, num_proc, user_parser,
```

Comment:
Have you implemented any logic to load the ext_mem objects as you flushed the ExtMemArrayWrapper._arr and set it into None.

Reply:
ExtNumpyWrapper loads the data from the file whenever data are accessed.